### ISSUE TITLE
[Feature Request] Edit Mode

### ISSUE THREAD

**Issue by RickyM7** [Automattic/simplenote-android#1169]
### What

1- Preview mode only (Markdown) when entering a note, instead of edit mode.

2- Edit mode in a separate option to make it more enjoyable to read and edit a note (It can be a pencil icon next to the information icon or just place the option where the markdown option is).

3- Markdown mode always active in edit mode for more options for personalizing notes.

4- Checklist items are only editable in edit mode, but check boxes are still marked in preview mode (or there could also be a "+" symbol at the end of the checklist to add a new item).

The button to enter edit mode would be where the "Markdown" button is currently located.
In edit mode, markdown is always enabled, so the options for customizing notes are greater.
Notes can only be edited within the edit mode, and outside the edit mode the notes would be basically as shown in the markdown preview, except for the checklist, as the check boxes must remain exactly as they are, and the check boxes can still be checked out of edit mode, however checklist items can only be edited in edit mode.

### Why

Edit mode would be something to improve the way notes are edited, this would avoid accidentally editing or writing notes and would improve notes customization and visualization.
This feature will be very useful and will make the app experience more enjoyable.

### How

Edit Mode Example:
![Edit Mode](https://user-images.githubusercontent.com/24703825/95616909-f89aa080-0a40-11eb-9b89-15171f239579.png)

Preview Mode Example:
![Preview Mode](https://user-images.githubusercontent.com/24703825/95616930-005a4500-0a41-11eb-8df2-96bc6d92ca4f.png)

***Note***: Examples from NextCloud Notes!

**Comment by theck13** [comment]
A note is not editable when viewing the **_Preview_** tab.  Therefore, opening a note in the **_Preview_** tab would be the same as the non-edit mode you described.  Simplenote remembers the last tab viewed with the note open.  That is on a per-note basis.  That means the note will open in edit mode if you viewed the **_Edit_** tab or non-edit mode if you viewed the **_Preview_** tab before exiting the note.  Also, Simplenote remembers whether markdown was last enabled or disabled and any new note will follow that setting.  That means a new note will have markdown enabled if the last note enabled markdown and disabled if the last note disabled markdown.  Additionally, if you accidentally edit a note, you can use the **_History_** to restore a previous version.

In the future, use the template for creating issues or the issue with be closed without comment.

**Comment by RickyM7** [comment]
@theck13 Got it, okay. Thanks for the answer.

**Comment by RickyM7** [comment]
> A note is not editable when viewing the **_Preview_** tab. Therefore, opening a note in the **_Preview_** tab would be the same as the non-edit mode you described.

But my request does not involve editing the notes in the markdown view mode, sorry if I couldn’t explain it properly, it’s because I’m using the google translator and the translation may have been wrong.
But I think I understand. Thanks for the answer.

**Comment by RickyM7** [comment]
@theck13 Could you take a look at this again?  I used the template to organize everything better.  I apologize if you are unable to understand something, is that I am using google translator to translate into english. In addition, there is no problem if this request cannot be made, after all this is just a complementary thing, I think.

**Comment by theck13** [comment]
No need to apologize.  Translators can be weird, but I think I understand your idea.

It looks like you want to replace the **_Edit_** and **_Preview_** tabs with an **_Edit_** and **_Markdown_** action.  Also, the checkbox rendering is the opposite of the current implementation in your proposal.

> 1- Preview mode only (Markdown) when entering a note, instead of edit mode.
> 
> 2- Edit mode in a separate option to make it more enjoyable to read and edit a note (It can be a pencil icon next to the information icon or just place the option where the markdown option is).

If users want to see the markdown preview when opening the note, they can view the **_Preview_** tab before exiting the note.  If users want to see the editable text when opening the note, they can view the **_Edit_** tab before exiting the note.  Effectively, that is the same as what you suggested, which I quoted above.  That's what I meant by my previous comment, which I quoted below.

> Simplenote remembers the last tab viewed with the note open. That is on a per-note basis. That means the note will open in edit mode if you viewed the **_Edit_** tab or non-edit mode if you viewed the **_Preview_** tab before exiting the note.

> 3- Markdown mode always active in edit mode for more options for personalizing notes.

The editor allows users to enter markdown text whether markdown is enabled or not.  The only difference is if markdown is not enabled, they can't preview what the markdown looks like in the app.  However, if they publish the not, it will still show the rendered markdown in the public link.

Not all users enable markdown.  Some of them just don't use it, because they don't want to.  If we enable markdown for _all_ users and always have an **_Edit_** and **_Markdown_** action, it will force another action from the top app bar into the overflow/ellipsis menu.  We have don't detailed analysis and testing to determine which actions are used the most often in order to make the actions in the top app bar the most convenient for the most users.

> 4- Checklist items are only editable in edit mode, but check boxes are still marked in preview mode (or there could also be a "+" symbol at the end of the checklist to add a new item).

As I mentioned above, not all users want to enable markdown.  That is why we render the checkbox and allow it to be toggled in edit mode.  Actually, we have debated if we should show `- [ ]` for unchecked items and `- [x]` for checked items in edit mode since that is the true markdown syntax and it would be more consistent with the rest of the edit mode text.  However, we decided that we want all users to be able to see the rendered checkboxes and toggle them in edit mode since not all users enable markdown.  We may change that in the future, but that's what we have decided at the moment.

As for not rendering the checkboxes in the markdown preview, that is a limitation of the markdown parser we're using.  Since Simplenote is very often used on multiple platforms by users, we want to have as much parity as possible between our apps.  In the case of our markdown parser, we chose the same base library for Android and iOS for consistency between those platforms.  Unfortunately, that means we have to make compromises, which limits our options in the markdown preview.

I believe I've addressed your updated points above.  Since there is a way for users to use the current Simplenote Android implementation the way you propose and we have some limitations with our markdown parser, I don't think we'll be making any changes.

Thanks for updating the issue with more details and screenshots!  Let me know if anything in my explanation in not clear.
